### PR TITLE
Add restaurant place manager

### PR DESCRIPTION
**Pull request by jyx11011**

### CODE REVIEW

File: src/test/java/com/google/step/datamanager/RestaurantPlaceManagerDatastoreTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  @Test
+  public void testUpdatePlacesOfRestaurant_morePlaceIds() {
+    List<String> placeIds = Arrays.asList(PLACE_ID_A, PLACE_ID_B);
+    restaurantPlaceManager.updatePlacesOfRestaurants(RESTAURANT_ID_A, placeIds);
+    placeIds = Arrays.asList(PLACE_ID_A, PLACE_ID_C, PLACE_ID_D);
+    restaurantPlaceManager.updatePlacesOfRestaurants(RESTAURANT_ID_A, placeIds);
+
+    Set<String> actual = restaurantPlaceManager.getPlaceIdsOfRestaurant(RESTAURANT_ID_A);
```
Comment (lisasgoh) [inline src/test/java/com/google/step/datamanager/RestaurantPlaceManagerDatastoreTest.java:55]: Maybe you could add an assert on the size of set

**Review by limli — DISMISSED** [review]

File: src/test/java/com/google/step/datamanager/RestaurantPlaceManagerDatastoreTest.java (hunk 1 above)
Comment (jyx11011) [reply src/test/java/com/google/step/datamanager/RestaurantPlaceManagerDatastoreTest.java:55]: ok I change hasItem to contains in any order, which checks size

File: src/main/java/com/google/step/datamanager/RestaurantPlaceManagerDatastore.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    for (Entity entity : results) {
+      String placeId = (String) entity.getProperty("placeId");
+      if (newPlaceIds.contains(placeId)) {
+        newPlaceIds.remove(placeId);
+        continue;
+      }
+      datastore.delete(entity.getKey());
+    }
+
+    // Adds remaining tags in the new list
```
Comment (limli) [inline src/main/java/com/google/step/datamanager/RestaurantPlaceManagerDatastore.java]:
```suggestion
    // Adds remaining place ids in the new list
```

**Review by limli — DISMISSED** [review]

File: src/main/java/com/google/step/datamanager/RestaurantPlaceManager.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.google.step.datamanager;
+
+import java.util.List;
+import java.util.Set;
+
+public interface RestaurantPlaceManager {
+  public void updatePlacesOfRestaurants(long restaurantId, List<String> placeIds);
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/datamanager/RestaurantPlaceManager.java]:
```suggestion
  public void updatePlacesOfRestaurant(long restaurantId, List<String> placeIds);
```

There's only a single restaurantId?

File: src/main/java/com/google/step/datamanager/RestaurantPlaceManager.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+  public Set<String> getPlaceIdsOfRestaurant(long id);
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/datamanager/RestaurantPlaceManager.java]:
```suggestion
  public Set<String> getPlaceIdsOfRestaurant(long restaurantId);
```

is clearer and disambiguates between the two IDs being referred to

File: src/main/java/com/google/step/datamanager/RestaurantPlaceManagerDatastore.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  }
+
+  private Entity createRestaurantPlaceEntity(long restaurantId, String placeId) {
+    Entity entity = new Entity("RestaurantPlace");
+    entity.setProperty("restaurantId", restaurantId);
+    entity.setProperty("placeId", placeId);
+    return entity;
+  }
+
+  private Iterable<Entity> getEntitiesOfRestaurant(long id) {
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/datamanager/RestaurantPlaceManagerDatastore.java]:
```suggestion
  private Iterable<Entity> getRestaurantPlaceEntities(long restaurantId) {
```

File: src/main/java/com/google/step/datamanager/RestaurantPlaceManagerDatastore.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    Query query = new Query("RestaurantPlace");
+    Filter filter = new FilterPredicate("restaurantId", FilterOperator.EQUAL, id);
+    query.setFilter(filter);
+    Iterable<Entity> results = datastore.prepare(query).asIterable();
+    return results;
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/datamanager/RestaurantPlaceManagerDatastore.java]:
```suggestion
    return datastore.prepare(query).asIterable();
```

File: src/main/java/com/google/step/datamanager/RestaurantPlaceManagerDatastore.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.Set;
+
+public class RestaurantPlaceManagerDatastore implements RestaurantPlaceManager {
+
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void updatePlacesOfRestaurants(long restaurantId, List<String> placeIds) {
+    Set<String> newPlaceIds = new HashSet<>(placeIds);
+    Iterable<Entity> results = getEntitiesOfRestaurant(restaurantId);
```
Comment (SuyashLakhotia) [inline src/main/java/com/google/step/datamanager/RestaurantPlaceManagerDatastore.java]:
```suggestion
    Iterable<Entity> currentPlaceEntities = getEntitiesOfRestaurant(restaurantId);
```

File: src/test/java/com/google/step/servlets/RestaurantServletTest.java [hunk 8]
```diff
@@ -65,7 +69,7 @@ public void testDoGet_success() throws Exception {
 
     String expected =
         String.format(
-            "{id:%d,name:\"%s\",photoUrl:\"%s\"," + "deals: []}",
+            "{id:%d,name:\"%s\",photoUrl:\"%s\",deals: [],placeIds:[]}",
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/servlets/RestaurantServletTest.java]: Add tests with non-empty deals & placeIds?

File: src/test/java/com/google/step/servlets/RestaurantPostServletTest.java [hunk 9]
```diff
@@ -60,7 +60,7 @@ public void testDoPost_success() throws Exception {
     restaurantPostServlet.doPost(mockRequest, mockResponse);
     String expected =
         String.format(
-            "{id:%d,name:\"%s\",photoUrl:\"%s\",deals:[]}",
+            "{id:%d,name:\"%s\",photoUrl:\"%s\",deals:[],placeIds:[]}",
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/servlets/RestaurantPostServletTest.java]: Add tests with non-empty deals & placeIds?

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]
For all method renames, please also rename the appropriate tests.

File: src/test/java/com/google/step/TestConstants.java [hunk 10]
```diff
@@ -116,15 +122,6 @@
...
-  // Deal Brief JSON
-  public static final String DEAL_A_BRIEF_JSON =
-      String.format(
-          "{restaurant: %d, description: \"%s\", votes: %d, id: %d, pic: \"%s\", poster: %d}",
-          RESTAURANT_ID_A, DESCRIPTION_A, VOTE_A, DEAL_ID_A, BLOBKEY_URL_A, USER_ID_A);
-  public static final String DEAL_B_BRIEF_JSON =
-      String.format(
-          "{restaurant: %d, description: \"%s\", votes: %d, id: %d, pic: \"%s\", poster: %d}",
-          RESTAURANT_ID_B, DESCRIPTION_B, VOTE_A, DEAL_ID_B, BLOBKEY_URL_B, USER_ID_B);
   // Deal
```
Comment (SuyashLakhotia) [inline src/test/java/com/google/step/TestConstants.java]: Delete this comment?

**Review by SuyashLakhotia — CHANGES_REQUESTED** [review]
Is this in sync with #103?

**Review by SuyashLakhotia — APPROVED** [review]
